### PR TITLE
Reservee name

### PR DESCRIPTION
**Pull request by ranta**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Extend Reservation.reservee_name with more options where to get the name
- Refactor some related old tests

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Automated tests

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-3022](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3022)

[TILA-3022]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3022?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: email_notification/sender/email_notification_context.py [hunk 1]
```diff
@@ -88,11 +87,7 @@ def from_reservation(reservation: Reservation) -> "EmailNotificationContext":
         """Build context from reservation"""
         context = EmailNotificationContext()
 
-        if not reservation.reservee_type or reservation.reservee_type == CustomerTypeChoice.INDIVIDUAL.value:
-            context.reservee_name = f"{reservation.reservee_first_name} {reservation.reservee_last_name}"
-        else:
-            context.reservee_name = reservation.reservee_organisation_name
-
+        context.reservee_name = reservation.reservee_name
```
Comment (vergama) [inline email_notification/sender/email_notification_context.py:90]: This is used in the salutation of the email text ("Hi FirstName LastName") so defaulting to reservation.name would be undesirable.

File: tests/test_graphql_api/test_reservation/test_query.py [hunk 2]
```diff
@@ -138,7 +138,7 @@ def test_reservation__query__all_fields(graphql):
         "reserveeId": reservation.reservee_id,
         "reserveeIsUnregisteredAssociation": reservation.reservee_is_unregistered_association,
         "reserveeLastName": reservation.reservee_last_name,
-        "reserveeName": f"{reservation.reservee_first_name} {reservation.reservee_last_name}",
+        "reserveeName": f"{reservation.user.first_name} {reservation.user.last_name}",
```
Comment (matti-lamppu) [inline tests/test_graphql_api/test_reservation/test_query.py:141]: `reservee_first_name` + `reservee_last_name` is not the same as `user.first_name` + `user.last_name` . The former names are saved to the reservation at the time of creation, which they might be changed after. @vergama can also check if this should be the case here.

Comment (ranta) [reply tests/test_graphql_api/test_reservation/test_query.py:141]: This was an intentional change, as the `reservee_type` here is `None`, so user's name is used as fallback when we don't know if the reservee is an individual or organisation (and also `reservation.name` is not defined)

File: reservations/models.py [hunk 3]
```diff
@@ -432,6 +432,40 @@ def requires_handling(self) -> bool:
...
 
+    @property
+    def reservee_name(self) -> str:
+        # Internal recurring reservations are created by STAFF
+        if self.type == ReservationTypeChoice.STAFF.value and self.recurring_reservation is not None:
+            return self.recurring_reservation.name
+
+        # Blocking reservation
+        if self.type == ReservationTypeChoice.BLOCKED.value:
+            return _("Closed")
```
Comment (vergama) [inline reservations/models.py:443]: Also need the locale message strings:
- fi: "Suljettu"
- sv: "Stängt"

File: email_notification/sender/email_notification_context.py (hunk 1 above)
Comment (ranta) [reply email_notification/sender/email_notification_context.py]: Changed these few rows back.

File: reservations/models.py (hunk 3 above)
Comment (ranta) [reply reservations/models.py:443]: Added translations for this, but something to consider is that we are currently missing ~99% of translations. :)

**Review by vergama — APPROVED** [review]

**Pull request merged**